### ISSUE TITLE
Tris working on Mac but not in Linux

### ISSUE THREAD

**Issue by MacTuitui** [nannou-org/nannou#427]
On my latest series I ended up with this strange bug where it works on Mac but not on Linux.

I'm still on nannou 0.11 and here is the code.

```rust
use nannou::noise::*;
use nannou::prelude::*;

struct Model {
    noise: Perlin,
}

fn main() {
    nannou::app(model).run();
}

fn model(app: &App) -> Model {
    let _window_id = app
        .new_window()
        .with_dimensions(1024, 1024)
        .view(view)
        .build()
        .unwrap();
    let mut noise = Perlin::new();
    noise = noise.set_seed(1 as u32);
    Model {
        noise,
    }
}

fn view(app: &App, model: &Model, frame: &Frame) {
    // Prepare to draw.
    let draw = app.draw();
    let  frac = (((app.elapsed_frames()) % 600) as f32) /600.0;

    let  c = BLACK;
    draw.background().color(c);

    let angle = frac*TAU;
    let acos = angle.cos() as f64;
    let asin = angle.sin() as f64;
    let steps_i = 10;
    let steps_j = 36;
    for i in 0..steps_i {
        let frac_i = ((i+1) as f32)/((steps_i) as f32);
        let mut points = Vec::new();
        for j in 0..steps_j {
            let frac_j = (j as f32)/((steps_j-1) as f32);
            let angle = frac_j *TAU;
            let v = model.noise.get([
                                    (angle.cos()*frac_i) as f64,
                                    (angle.sin()*frac_i) as f64,
                                    acos*1.0,
                                    asin*1.0]) as f32
                                    *TAU;
           let p = vec2(angle.cos()*500.0*frac_i+(1.0-frac_i)*200.0*v.cos(), 
angle.sin()*500.0*frac_i + (1.0-frac_i)*200.0*v.sin());
            points.push(p);
        }
        let  clear:Srgba = srgba(0.0,1.0,0.0, 0.0);//(BLACK,1.0);//.into();
        let  opaque:Srgba = srgba(1.0,0.0,0.0, 1.0);//(BLACK,1.0);//.into();

        let tris = (1..points.len()).flat_map( |i| {
            let p0 = &points[i-1];
            let p1 = &points[i];
            let l = p0.magnitude();
            let la = l+80.0*(1.0-frac_i)+2.0;
            let l = p1.magnitude();
            let lb = l+80.0*(1.0-frac_i)+2.0;
            let a = geom::vertex::Srgba(vec2(p0.x, p0.y), opaque);
            let b = geom::vertex::Srgba(vec2(p1.x, p1.y), opaque);
            let c = geom::vertex::Srgba(vec2(p1.x, p1.y).normalize()*lb, clear);
            let d = geom::vertex::Srgba(vec2(p0.x, p0.y).normalize()*la, clear);

            geom::Quad([a,b,c,d]).triangles_iter()
        });
        draw.mesh().tris(tris);
        let clear:Srgba = srgba(0.0,0.0,0.0, 0.0);//(BLACK,1.0);//.into();
        let tris = (1..points.len()).flat_map( |i| {
            let p0 = &points[i-1];
            let p1 = &points[i];
            let l = p0.magnitude();
            let la = l-80.0*(1.0-frac_i)+2.0;
            let l = p1.magnitude();
            let lb = l-80.0*(1.0-frac_i)+2.0;
            let a = geom::vertex::Srgba(vec2(p0.x, p0.y), opaque);
            let b = geom::vertex::Srgba(vec2(p1.x, p1.y), opaque);
            let c = geom::vertex::Srgba(vec2(p1.x, p1.y).normalize()*lb, clear);
            let d = geom::vertex::Srgba(vec2(p0.x, p0.y).normalize()*la, clear);

            geom::Quad([a,b,c,d]).triangles_iter()
        });
        draw.mesh().tris(tris);
    }

    draw.to_frame(app, &frame).unwrap();

}
```
So I'm basically drawing `steps_i` times two sets of triangles, using noise to displace them.
On Mac I get the expected result:
<img width="1020" alt="mac" src="https://user-images.githubusercontent.com/1924395/68825028-17cb6980-06dc-11ea-9dd3-e1cc9dbf7dc2.png">

But on Linux I have the following (screen shot not taken at the same time, but it only shows the *first* set of tris):
![linux-test](https://user-images.githubusercontent.com/1924395/68825050-2580ef00-06dc-11ea-8d3b-35385217787e.png)

Now I have no idea what to do to debug this...

**Comment by freesig** [comment]
Do the nannou examples work on your Linux?

**Comment by MacTuitui** [comment]
They actually do, and in the same repository as well (by copying an example to my rep).
(The fact that the triangles should also be rendered with transparent vertices suggests another issue that might be related?)

**Comment by MacTuitui** [comment]
Fixed on nannou 0.12 / vulkano 0.16 actually, so can't really know what was wrong.
